ingestion-beam/src/main/java/com/mozilla/telemetry/util/DynamicPathTemplate.java

```diff
@@ -100,13 +100,7 @@ public String replaceDynamicPart(List<String> values) {
       String name = placeholderNames.get(i);
       String value = attributes.get(name);
       if (value == null) {
-        String defaultValue = placeholderDefaults.get(i);
-        if (defaultValue == null) {
-          throw new IllegalArgumentException("The destination specified via --output"
-              + " includes an attribute placeholder `" + name + "' with no default,"
-              + " so a value is required, but this message has no attribute `" + name + "'.");
-        }
-        value = defaultValue;
```

Comment:
Since changing to requiring defaults, this situation can never occur, so this was dead code.